c7n/query.py

```diff
@@ -540,6 +529,58 @@ def generate_arn(self):
         return self._generate_arn
 
 
+class MaxResourceLimit(object):
+
+    C7N_MAXRES_OP = os.environ.get("C7N_MAXRES_OP", 'or')
+    C7N_MAXRES_PERCENT = os.environ.get("C7N_MAXRES_PERCENT", 100)
+
+    def __init__(self, policy, selection_count, population_count):
+        self.p = policy
+        self.op = MaxResourceLimit.C7N_MAXRES_OP
+        self.selection_count = selection_count
+        self.population_count = population_count
+        self.amount = selection_count
+        self.percentage_amount = population_count
+        self.percent = MaxResourceLimit.C7N_MAXRES_PERCENT
+        self._parse_policy()
+
+    def _parse_policy(self,):
+        if isinstance(self.p.max_resources, dict):
+            self.op = self.p.max_resources.get("op", MaxResourceLimit.C7N_MAXRES_OP).lower()
+            self.percent = self.p.max_resources.get("percent",
+                                                    MaxResourceLimit.C7N_MAXRES_PERCENT)
+            self.amount = self.p.max_resources.get("amount", self.population_count)
+
+        if isinstance(self.p.max_resources, int):
+            self.amount = self.p.max_resources
+
+        if isinstance(self.p.max_resources_percent, (int, float)):
+            self.percent = self.p.max_resources_percent
+
+        self.percentage_amount = self.population_count * (self.percent / 100.0)
+
+    def check_resource_limits(self):
+        if self.selection_count > self.amount and self.selection_count > self.percentage_amount \
```

Comment:
style minor nit: rather than line continuation \ use ( ) around the conditional to break it across multiple lines.